frontend/packages/dev-console/src/components/monitoring/alerts/MonitoringAlerts.tsx

```diff
@@ -0,0 +1,161 @@
+import { FilterToolbar } from '@console/internal/components/filter-toolbar';
+import { getAlertsAndRules } from '@console/internal/components/monitoring/utils';
+import { monitoringSetRules, sortList } from '@console/internal/actions/ui';
+import { useURLPoll } from '@console/internal/components/utils/url-poll-hook';
+import { PrometheusRulesResponse, Rules } from '@console/internal/components/monitoring/types';
+import { PROMETHEUS_TENANCY_BASE_PATH } from '@console/internal/components/graphs';
+import { RootState } from '@console/internal/redux';
+import { alertingRuleStateOrder } from '@console/internal/reducers/monitoring';
+import { getFilteredRows } from '@console/internal/components/factory';
+import {
+  monitoringAlertRows,
+  monitoringAlertColumn,
+  alertFilters,
+  applyListSort,
+} from './monitoring-alerts-utils';
+import './MonitoringAlerts.scss';
+import { getURLSearchParams } from '@console/internal/components/utils';
+
+type MonitoringAlertsProps = {
+  match: RMatch<{
+    ns?: string;
+  }>;
+};
+
+type StateProps = {
+  rules: Rules;
+  filters: { [key: string]: any };
+  listSorts: any;
+};
+
+type props = MonitoringAlertsProps & StateProps;
+
+const POLL_DELAY = 15 * 1000;
+const reduxID = 'devMonitoringAlerts';
+const textFilter = 'resource-list-text';
+
+export const MonitoringAlerts: React.FC<props> = ({ match, rules, filters, listSorts }) => {
+  const [sortBy, setSortBy] = React.useState({});
+  const [rows, setRows] = React.useState([]);
+  const [collapsedRows, setCollapsedRows] = React.useState([]);
+  const dispatch = useDispatch();
+  const namespace = match.params.ns;
+  const { sortBy: listSortBy, orderBy: listOrderBy } = getURLSearchParams();
+  const columnIndex = _.findIndex(monitoringAlertColumn, { title: listSortBy });
+  const sortOrder = listOrderBy || SortByDirection.asc;
+  const [response, loadError, loading] = useURLPoll<PrometheusRulesResponse>(
+    `${PROMETHEUS_TENANCY_BASE_PATH}/api/v1/rules?namespace=${namespace}`,
+    POLL_DELAY,
+    namespace,
+  );
+  const thanosRules = React.useMemo(
+    () => (!loading && !loadError ? getAlertsAndRules(response?.data).rules : []),
+    [response, loadError, loading],
+  );
+
+  React.useEffect(() => {
+    const sortThanosRules = _.sortBy(thanosRules, (rule) => alertingRuleStateOrder(rule));
+    dispatch(monitoringSetRules('devRules', sortThanosRules));
+  }, [dispatch, thanosRules]);
+
+  const filteredRules = React.useMemo(() => {
+    if (columnIndex > -1) {
+      setSortBy({ index: columnIndex + 1, direction: sortOrder });
+    }
+    const fRules = getFilteredRows(filters, alertFilters, rules);
+    const { field, orderBy } = listSorts;
+    return applyListSort(fRules, orderBy || listOrderBy, field || _.camelCase(listSortBy));
+  }, [listSorts, columnIndex, sortOrder, filters, rules, listOrderBy, listSortBy]);
+
+  React.useEffect(() => {
+    const tableRows = monitoringAlertRows(filteredRules);
```

Comment:
It would be more efficient and simpler to pass `collapsedRows` to `monitoringAlertRows` (after changing `collapsedRows` to an array of row IDs). Then this would just be `const tableRows = monitoringAlertRows(sortedRules, collapsedRows)`.